### ISSUE TITLE
ERROR non 200 status code received error=<nil>

### ISSUE THREAD

**Issue by Rak3tenmann** [kevinzehnder/enex2paperless#1]
Hi there!
First of all: Thanks for your effort by writing this tool.
Very im,portant stuff to migrate.

I just trried, and I am sure, this is my fault. 

Config file is
PaperlessAPI: http://192.168.5.227:8010/
Username: admin
Password: supersecretpw

Output:
2024/01/29 21:45:26 INFO uploading file file=Inter-19022020.pdf
2024/01/29 21:45:26 ERROR non 200 status code received error=<nil>
2024/01/29 21:45:26 INFO ENEX processing done numberOfNotes=5 totalUploads=0

Thank you!

**Comment by kevinzehnder** [comment]
I cant verify right now, but Id assume its the trailing slash in the API URL. Remove that and see what happens. 

But it shows that there should be better error messages. Not a lot of info here except that the commection isnt working as expected.

**Comment by kevinzehnder** [comment]
It's probably something else. Could be the wrong credentials? download v0.0.2 and try. It should give more information about the error.

**Comment by Rak3tenmann** [comment]
I have just seen that my system has deleted your exe because it suspects a virus.

**Comment by kevinzehnder** [comment]
Yea that is sort of expected. I don't have any valid code-signing certificates, so this code will look suspicious and untrusted. As it's often the case with small open-source projects. You can read the source code, so there's that :)

**Comment by Rak3tenmann** [comment]
Fine with it - and I know such cases of course. But in this case, it was much harder stop. I came to my computer, and the exe was gone. 
Wasn't easy but fixed it. Also, you are right about the slash. However, received a differnt error now:

2024/01/30 15:12:49 INFO uploading file file=2020_08_28_06_35_52.pdf
2024/01/30 15:12:49 ERROR error creating new HTTP request error="parse \"192.168.5.227:8010/api/documents/post_document/\": first path segment in URL cannot contain colon"

Hope it's ok to keep thisd in the same issue here.

**Comment by kevinzehnder** [comment]
Sure, lets keep it in this thread. Its all related to the URL. I'll go over it and try to add some validation for the config, to avoid something like this.

**Comment by Rak3tenmann** [comment]
But how should I avoid colon? Paparless in in da Docker COntainer. I need to mention the port here.

**Comment by kevinzehnder** [comment]
that's probably because you're missing the "http://" part in the config file. Try version 0.0.3, that will verify that the config has a correct HTTP URL and avoid problems like these. If you change to
```
PaperlessAPI: http://192.168.5.227:8010
Username: admin
Password: supersecretpw
```
it should work

**Comment by Rak3tenmann** [comment]
Now I got
C:\Users\Flow\Desktop\enex2paperless>enex2paperless 1.enex
2024/01/31 14:22:48 INFO uploading file file=2020_08_28_06_35_52.pdf
2024/01/31 14:22:50 ERROR non 200 status code received "status code"=401
2024/01/31 14:22:50 ERROR response: body="{\"detail\":\"Invalid username/password.\"}"

Double checked login data. 
:-(

**Comment by kevinzehnder** [comment]
Could be some YAML peculiarity. Check for whitespaces and special characters like ' / or : 
YAML also treats these things differently:
```
Password: hello
Password: "hello"
Password: 'hello'
```

**Comment by Rak3tenmann** [comment]
without the whitespoaces I get another error
C:\Users\Flow\Desktop\enex2paperless>enex2paperless 1.enex
2024/01/31 15:28:33 ERROR couldn't read config error="configuration error: Key: 'Config.PaperlessAPI' Error:Field validation for 'PaperlessAPI' failed on the 'required' tag\nKey: 'Config.Username' Error:Field validation for 'Username' failed on the 'required' tag\nKey: 'Config.Password' Error:Field validation for 'Password' failed on the 'required' tag"


Tried with " and ' 
wrong PW or username.

**Comment by kevinzehnder** [comment]
If you get that validation error it means that the program wasnt able to read a username or password from the file at all. 

Whitespace is necessary but make sure theres no additional whitespace at the end or beginning for instance. Can you add the config.yaml to this post?

**Comment by Rak3tenmann** [comment]
[config.zip](https://github.com/kevinzehnder/enex2paperless/files/14114419/config.zip)
here you go

**Comment by kevinzehnder** [comment]
I really can't see what's wrong. Can you check the Paperless logs? 
Do you have a lot of these entries?
```
[paperless.auth] Login failed for user `user` from private IP `x.x.x.x`.
```

**Comment by Rak3tenmann** [comment]
Good point - yes - I do. But user "Flow" 

[2024-01-31 18:25:15,429] [INFO] [paperless.auth] Login failed for user `Flow` from private IP `192.168.5.225`.
Any clue?

I also tried to login via Browser with a wrong PW.
Here, the Logs are correct.

[2024-01-31 18:29:57,684] [INFO] [paperless.auth] Login failed for user `admin` from private IP `192.168.5.225`.

But over API - it always comes back with User 'Flow' doesn't matter whatg name I use in the Config FIle. Even if I leave it blank.

**Comment by kevinzehnder** [comment]
that is very strange and I don't really know what's going on. Where does "Flow" come from? Maybe try to create a new user and use that one?

**Comment by Rak3tenmann** [comment]
Haha.
Found a way :-) Guess what: If I setup a user named "Flow" it works.
Well - this is my workarround for now.

**Comment by kevinzehnder** [comment]
If it works, it works :)

**Comment by Rak3tenmann** [comment]
Since I need it just once - i am fine with it.
However, I will try to migrate about 3k objects tomorrow. Will let you know....

**Comment by kevinzehnder** [comment]
Did you run into more problems or were you able to let it run?

**Comment by hiddenpcmaster** [comment]
I'm not sure if this is related or not ......

2024/02/04 11:48:44 INFO uploading file file=2023_10_21_17_19_34.pdf
2024/02/04 11:48:44 ERROR non 200 status code received "status code"=403
2024/02/04 11:48:44 ERROR response: body="\n<!DOCTYPE html>\n<html lang=\"en\">\n<head>\n  <meta http-equiv=\"content-type\" content=\"text/html; charset=utf-8\">\n  <meta name=\"robots\" content=\"NONE,NOARCHIVE\">\n  <title>403 Forbidden</title>\n  <style type=\"text/css\">\n    html * { padding:0; margin:0; }\n    body * { padding:10px 20px; }\n    body * * { padding:0; }\n    body { font:small sans-serif; background:#eee; color:#000; }\n    body>div { border-bottom:1px solid #ddd; }\n    h1 { font-weight:normal; margin-bottom:.4em; }\n    h1 span { font-size:60%; color:#666; font-weight:normal; }\n    #info { background:#f6f6f6; }\n    #info ul { margin: 0.5em 4em; }\n    #info p, #summary p { padding-top:10px; }\n    #summary { background: #ffc; }\n    #explanation { background:#eee; border-bottom: 0px none; }\n  </style>\n</head>\n<body>\n<div id=\"summary\">\n  <h1>Forbidden <span>(403)</span></h1>\n  <p>CSRF verification failed. Request aborted.</p>\n\n\n  <p>You are seeing this message because this site requires a CSRF cookie when submitting forms. This cookie is required for security reasons, to ensure that your browser is not being hijacked by third parties.</p>\n  <p>If you have configured your browser to disable cookies, please re-enable them, at least for this site, or for “same-origin” requests.</p>\n\n</div>\n\n<div id=\"explanation\">\n  <p><small>More information is available with DEBUG=True.</small></p>\n</div>\n\n</body>\n</html>\n"

**Comment by kevinzehnder** [comment]
could be a general paperless problem? Did you set PAPERLESS_URL ?

**Comment by hiddenpcmaster** [comment]
As a matter of fact, yes.  This field is set with my domain that hits my reverse proxy.  I appreciate you possibly sending me in the right direction.  I may just try reverting to a local instance, at least for the import.  Will let you know how it goes.  Appreciate your efforts regardless.

**Comment by giukanis** [comment]
> that is very strange and I don't really know what's going on. Where does "Flow" come from? Maybe try to create a new user and use that one?

Hi, I had the same issue after deleting the trailing slash from the URL. The script wanted to login with another user. Seems to be something like the windows user perhaps. If there is a trailing slash in the url the user is picked correctly...

For me it worked also to change the username in paperless to the (windows?) username.

**Comment by kevinzehnder** [comment]
I assume the username problem is fixed with v0.0.4
